service/collector_test.go

```diff
@@ -106,13 +107,17 @@ func testCollectorStartHelper(t *testing.T) {
 	}
 
 	metricsPort := testutil.GetAvailablePort(t)
-	cfgProvider, err := NewConfigProvider(
-		[]string{filepath.Join("testdata", "otelcol-config.yaml")},
-		WithConfigMapConverters([]config.MapConverterFunc{
-			configmapprovider.NewOverwritePropertiesConverter(
-				[]string{"service.telemetry.metrics.address=localhost:" + strconv.FormatUint(uint64(metricsPort), 10)},
-			),
-		}))
+	cfgSet := NewDefaultConfigProviderSettings()
+	cfgSet.Locations = []string{
```

Comment:
I didn't know this before, but I prefer this so much more than the previous iteration.